### PR TITLE
Fix autoplay rewind/skip usable when UI hidden

### LINKED ISSUE #4127: Autoplay rewind/skip clickable when UI is hidden (shift+tab)
**Describe your problem:** When the UI is hidden with shift+tab the rewind/skip bar on the bottom can still be clicked on and works even though it is invisible.

**Screenshots or videos showing encountered issue:**  [video](https://puu.sh/CANJM/1a6d03722e.mp4)

**osu!lazer version:** 2019.122.1

### PR DESCRIPTION
**Pull request by chrisny286**
- Closes #4127 

---

Fix song progress bar clickable when UI is hidden

### CODE REVIEW

File: osu.Game/Screens/Play/HUDOverlay.cs [hunk 1]
```diff
@@ -102,6 +102,7 @@ private void load(OsuConfigManager config, NotificationOverlay notificationOverl
         {
             showHud = config.GetBindable<bool>(OsuSetting.ShowInterface);
             showHud.ValueChanged += hudVisibility => visibilityContainer.FadeTo(hudVisibility ? 1 : 0, duration);
+            showHud.ValueChanged += v => Progress.AllowSeeking = v;
```
Comment (ekrctb) [inline osu.Game/Screens/Play/HUDOverlay.cs]: This seems wrong when `AllowSeeking` was false and hud made visible.

**Review by ekrctb — CHANGES_REQUESTED** [review]
The problem is `visibilityContainer` is `AlwaysVisible` and propagates input when it is not visible. Could `AlwaysVisible` be removed? I'm not sure which "may need to still be updated" so needs to be investigated and probably `ProxyDrawable` is used.

File: osu.Game/Screens/Play/HUDOverlay.cs (hunk 1 above)
Comment (ekrctb) [reply osu.Game/Screens/Play/HUDOverlay.cs]: Also there is no reason to separate to two event additions.

Comment (chrisny286) [reply osu.Game/Screens/Play/HUDOverlay.cs]: yea thats right its only allowed to change when a replay / autoplay is loaded

File: osu.Game/Screens/Play/SongProgress.cs [hunk 2]
```diff
@@ -125,7 +125,7 @@ public bool AllowSeeking
             set
             {
                 if (allowSeeking == value) return;
-
+                if (!replayLoaded) return;
```
Comment (ekrctb) [inline osu.Game/Screens/Play/SongProgress.cs]: This change is completely unrelated. Please look at my review summary comment.

File: osu.Game/Screens/Play/HUDOverlay.cs (hunk 1 above)
Comment (chrisny286) [reply osu.Game/Screens/Play/HUDOverlay.cs]: ok so i didnt find any issue making the visibility container not always present.
fixes the problem as well.

Comment (chrisny286) [reply osu.Game/Screens/Play/HUDOverlay.cs]: since the updates the elements in the `visibilityContainer` receive are always from sources which are not triggered through Positional or NonPositional input.

Comment (ekrctb) [reply osu.Game/Screens/Play/HUDOverlay.cs]: git-blame revealed #3762 was the issue. Are you sure it is not regressed?

Comment (chrisny286) [reply osu.Game/Screens/Play/HUDOverlay.cs]: The Hold to quit button is in a different `FillFlowContainer` and is not Hidden by shift+tab

File: osu.Game/Screens/Play/SongProgress.cs [hunk 3]
```diff
@@ -125,7 +125,6 @@ public bool AllowSeeking
             set
             {
                 if (allowSeeking == value) return;
-
```
Comment (ekrctb) [inline osu.Game/Screens/Play/SongProgress.cs]: revert (empty line)

File: osu.Game/Screens/Play/HUDOverlay.cs (hunk 1 above)
Comment (chrisny286) [reply osu.Game/Screens/Play/HUDOverlay.cs]: None of the elements in the `visibilityContainer` need to  receive input at any time except the `Progress` which is only supposed to accept it when visible.

Comment (ekrctb) [reply osu.Game/Screens/Play/HUDOverlay.cs]: Actually, `ModDisplay` -> `ModIcon` receives positional input to display a tooltip but it works fine with this change. Confirmed.

**Review by ekrctb — APPROVED** [review]
